Search bar not working, keeps crashing

![image](https://github.com/vvbbnn00/onedrive-index/assets/144337960/e2b87e9b-a77b-4553-9a93-77058b98f866)

framework-de9232d192263565.js:9 TypeError: Cannot read properties of undefined (reading 'slice')
    at getRawExtension (5349-bd8d89f4684b8374.js:1:70789)
    at getExtension (5349-bd8d89f4684b8374.js:1:70857)
    at getFileIcon (5349-bd8d89f4684b8374.js:1:70922)
    at SearchResultItemTemplate (5349-bd8d89f4684b8374.js:1:52390)
    at Xh (framework-de9232d192263565.js:9:60992)
    at x (framework-de9232d192263565.js:9:119447)
    at Vk (framework-de9232d192263565.js:9:99132)
    at framework-de9232d192263565.js:9:98999
    at Jk (framework-de9232d192263565.js:9:99006)
    at Ok (framework-de9232d192263565.js:9:95762)
Mi @ framework-de9232d192263565.js:9
main-cf6e1d8c39f3c97e.js:1 TypeError: Cannot read properties of undefined (reading 'slice')
    at getRawExtension (5349-bd8d89f4684b8374.js:1:70789)
    at getExtension (5349-bd8d89f4684b8374.js:1:70857)
    at getFileIcon (5349-bd8d89f4684b8374.js:1:70922)
    at SearchResultItemTemplate (5349-bd8d89f4684b8374.js:1:52390)
    at Xh (framework-de9232d192263565.js:9:60992)
    at x (framework-de9232d192263565.js:9:119447)
    at Vk (framework-de9232d192263565.js:9:99132)
    at framework-de9232d192263565.js:9:98999
    at Jk (framework-de9232d192263565.js:9:99006)
    at Ok (framework-de9232d192263565.js:9:95762)
renderError @ main-cf6e1d8c39f3c97e.js:1
main-cf6e1d8c39f3c97e.js:1 A client-side exception has occurred, see here for more info: https://nextjs.org/docs/messages/client-side-exception-occurred
renderError @ main-cf6e1d8c39f3c97e.js:1
framework-de9232d192263565.js:9 TypeError: Cannot read properties of undefined (reading 'slice')
    at getRawExtension (5349-bd8d89f4684b8374.js:1:70789)
    at getExtension (5349-bd8d89f4684b8374.js:1:70857)
    at getFileIcon (5349-bd8d89f4684b8374.js:1:70922)
    at SearchResultItemTemplate (5349-bd8d89f4684b8374.js:1:52390)
    at Xh (framework-de9232d192263565.js:9:60992)
    at x (framework-de9232d192263565.js:9:119447)
    at Vk (framework-de9232d192263565.js:9:99132)
    at framework-de9232d192263565.js:9:98999
    at Jk (framework-de9232d192263565.js:9:99006)
    at Ok (framework-de9232d192263565.js:9:95762)
Mi @ framework-de9232d192263565.js:9
main-cf6e1d8c39f3c97e.js:1 TypeError: Cannot read properties of undefined (reading 'slice')
    at getRawExtension (5349-bd8d89f4684b8374.js:1:70789)
    at getExtension (5349-bd8d89f4684b8374.js:1:70857)
    at getFileIcon (5349-bd8d89f4684b8374.js:1:70922)
    at SearchResultItemTemplate (5349-bd8d89f4684b8374.js:1:52390)
    at Xh (framework-de9232d192263565.js:9:60992)
    at x (framework-de9232d192263565.js:9:119447)
    at Vk (framework-de9232d192263565.js:9:99132)
    at framework-de9232d192263565.js:9:98999
    at Jk (framework-de9232d192263565.js:9:99006)
    at Ok (framework-de9232d192263565.js:9:95762)
renderError @ main-cf6e1d8c39f3c97e.js:1
main-cf6e1d8c39f3c97e.js:1 A client-side exception has occurred, see here for more info: https://nextjs.org/docs/messages/client-side-exception-occurred
renderError @ main-cf6e1d8c39f3c97e.js:1
main-cf6e1d8c39f3c97e.js:1 Error rendering page:  Error: Cancel rendering route
    at _ (main-cf6e1d8c39f3c97e.js:1:13997)
    at main-cf6e1d8c39f3c97e.js:1:13949
    at new Promise (<anonymous>)
    at doRender (main-cf6e1d8c39f3c97e.js:1:13926)
    at main-cf6e1d8c39f3c97e.js:1:12063
(anonymous) @ main-cf6e1d8c39f3c97e.js:1

--- a/app/src/pages/api/item.ts
+++ b/app/src/pages/api/item.ts
@@ -97,7 +97,7 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
         try {
             queryId = decryptData(id);
             // console.log(queryId);
-            assert(queryId.match(/^[A-Za-z0-9]+$/))
+            assert(queryId.match(/^[A-Za-z0-9!]+$/))
         } catch (err) {
             res.status(400).json({error: 'Invalid driveItem ID.'})
             return;

--- a/app/src/pages/api/search.ts
+++ b/app/src/pages/api/search.ts
@@ -62,6 +62,7 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
         delete item?.parentReference?.driveId
         delete item?.parentReference?.driveType
         delete item?.parentReference?.siteId
+        delete item?.parentReference?.path
         item.id = encryptData(item.id);
         if (item.parentReference.id) {
           item.parentReference.id = encryptData(item?.parentReference?.id)

--- a/app/src/pages/setup/step-1.tsx
+++ b/app/src/pages/setup/step-1.tsx
@@ -50,7 +50,8 @@ export default function OAuthStep1({ installed, apiConfig, build_id }) {
   const router = useRouter()
   if (installed) {
     router.query.path = router.pathname.substring(1).split('/')
-    return Folders(build_id)
+    //@ts-ignore
+    return Folders({ build_id })
   }
 
   return (

--- a/app/src/pages/setup/step-2.tsx
+++ b/app/src/pages/setup/step-2.tsx
@@ -51,7 +51,8 @@ export default function OAuthStep2({ installed, redirectUri, oAuthUrl, build_id 
 
   if (installed) {
     router.query.path = router.pathname.substring(1).split('/')
-    return Folders(build_id)
+    //@ts-ignore
+    return Folders({ build_id })
   }
 
   return (

--- a/app/src/pages/setup/step-3.tsx
+++ b/app/src/pages/setup/step-3.tsx
@@ -74,7 +74,8 @@ export default function OAuthStep3({ accessToken, expiryTime, refreshToken, erro
 
   if (installed) {
     router.query.path = router.pathname.substring(1).split('/')
-    return Folders(build_id)
+    //@ts-ignore
+    return Folders({ build_id })
   }
 
   const sendAuthTokensToServer = async () => {

--- a/app/src/utils/getFileIcon.ts
+++ b/app/src/utils/getFileIcon.ts
@@ -14,7 +14,7 @@ const icons: { [key: string]: [IconPrefix, IconName] } = {
   file: ['far', 'file'],
   markdown: ['fab', 'markdown'],
   book: ['fas', 'book'],
-  link: ['fas', 'link'],
+  link: ['fas', 'link']
 }
 
 const extensions = {
@@ -92,7 +92,7 @@ const extensions = {
   mobi: icons.book,
   azw3: icons.book,
 
-  url: icons.link,
+  url: icons.link
 }
 
 /**
@@ -109,9 +109,13 @@ export function hasKey(obj: Record<string, any>, key: string): boolean {
   return key in obj
 }
 
-export function getRawExtension(fileName: string): string {
+export function getRawExtension(fileName: string | undefined): string {
+  if (typeof fileName !== 'string') {
+    return ''
+  }
   return fileName.slice(((fileName.lastIndexOf('.') - 1) >>> 0) + 2)
 }
+
 export function getExtension(fileName: string): string {
   return getRawExtension(fileName).toLowerCase()
 }

